qiskit/pulse/interfaces.py

```diff
@@ -11,127 +11,26 @@
 # that they have been altered from the originals.
 
 """
-ScheduleComponent, a common interface for components of schedule (Instruction and Schedule).
+NamedValue, a common interface for components of schedule (Instruction and Schedule) and beyond.
 """
 from abc import ABCMeta, abstractmethod
-from typing import Tuple, List, Union, Optional
-
+from typing import List
 from qiskit.pulse.channels import Channel
 
 # pylint: disable=missing-type-doc
 
 
-class ScheduleComponent(metaclass=ABCMeta):
-    """Common interface for components of schedule. """
+class NamedValue(metaclass=ABCMeta):
```

Comment:
Could we move this to a module `values`? and keep the old `ScheduleComponent` but raise a deprecation warning if imported?

Reply:
Should I open a separate PR for the changes or commit on top of this ?